### PR TITLE
fix(sentry): Restore sentry

### PR DESCRIPTION
**Pull request by michalszorad**
Resolves [YOMO-680](https://emurgo.atlassian.net/jira/software/c/projects/YOMO/issues/YOMO-680)

It has turned out sentry was not configured here at all

[YOMO-680]: https://emurgo.atlassian.net/browse/YOMO-680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/package.json [hunk 1]
```diff
@@ -135,6 +136,7 @@
     "expo": ">=48.0.0-0 <49.0.0",
     "expo-barcode-scanner": "~12.3.2",
     "expo-camera": "^13.2.1",
+    "expo-device": "^5.4.0",
```
Comment (michalszorad) [inline apps/wallet-mobile/package.json:140]: required by `sentry-expo`

File: apps/wallet-mobile/package.json [hunk 2]
```diff
@@ -169,6 +171,7 @@
     "react-native-webview": "^11.25.0",
     "react-query": "^3.39.3",
     "reselect": "^4.0.0",
+    "sentry-expo": "^7.0.1",
```
Comment (michalszorad) [inline apps/wallet-mobile/package.json:175]: Recommended way to use sentry on the expo project. Also fixes an issue where unresolved promises are not captured

File: apps/wallet-mobile/src/InitApp.tsx [hunk 3]
```diff
@@ -50,3 +57,23 @@ export const initApp = async (storage: YoroiStorage) => {
...
+    if (!CONFIG.SENTRY.ENABLE || !isString(CONFIG.SENTRY.DSN)) return
+    Sentry.init({
+      dsn: CONFIG.SENTRY.DSN,
+      patchGlobalPromise: true,
+      enableInExpoDevelopment: false,
+      tracesSampleRate: 1.0,
+      beforeSend(event) {
+        // https://github.com/getsentry/sentry-javascript/issues/2039
+        const isEnabled = ref.current
+        return isEnabled ? event : null
```
Comment (michalszorad) [inline apps/wallet-mobile/src/InitApp.tsx:75]: Drops all events when the user opts out from crash reports - https://github.com/getsentry/sentry-javascript/issues/2039#issuecomment-519439950

File: apps/wallet-mobile/src/features/Settings/ApplicationSettings/ApplicationSettingsScreen.tsx [hunk 4]
```diff
@@ -120,11 +118,7 @@ export const ApplicationSettingsScreen = () => {
             label={strings.crashReporting}
             info={strings.crashReportingInfo}
           >
-            <Switch
-              value={crashReports.enabled}
-              onValueChange={crashReports.enabled ? crashReports.disable : crashReports.enable}
-              disabled={isNightly() || isTogglePrivacyModeLoading}
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Settings/ApplicationSettings/ApplicationSettingsScreen.tsx:126]: @stackchain is there a reason we forced crash reports on nightly? Please let me know if I need to bring this back.

File: apps/wallet-mobile/src/legacy/config.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ export const isNightly = () => BUILD_VARIANT === 'NIGHTLY'
 // TODO(v-almonacid): consider adding 'ENABLE' as an env variable
 const SENTRY = {
   DSN: env.getString('SENTRY'),
-  ENABLE: __DEV__ || BUILD_VARIANT === 'NIGHTLY',
+  ENABLE: true,
```
Comment (michalszorad) [inline apps/wallet-mobile/src/legacy/config.ts]: Isn't this obsolete? Why was it enabled only on dev and nightly previously?

File: apps/wallet-mobile/src/InitApp.tsx [hunk 6]
```diff
@@ -50,3 +57,23 @@ export const initApp = async (storage: YoroiStorage) => {
...
+const useInitSentry = (options: {enabled: boolean}) => {
+  const ref = useRef(options.enabled)
+  ref.current = options.enabled
+
+  useEffect(() => {
+    if (!CONFIG.SENTRY.ENABLE || !isString(CONFIG.SENTRY.DSN)) return
+    Sentry.init({
+      dsn: CONFIG.SENTRY.DSN,
+      patchGlobalPromise: true,
+      enableInExpoDevelopment: false,
```
Comment (michalszorad) [inline apps/wallet-mobile/src/InitApp.tsx]: Do we want to track events in dev?

File: apps/wallet-mobile/src/features/Settings/ApplicationSettings/ApplicationSettingsScreen.tsx (hunk 4 above)
Comment (stackchain) [reply apps/wallet-mobile/src/features/Settings/ApplicationSettings/ApplicationSettingsScreen.tsx:126]: yes, on nightly we don't allow users to opt out, the whole point of nightly is to capture the issues before hitting prod

File: apps/wallet-mobile/src/legacy/config.ts (hunk 5 above)
Comment (stackchain) [reply apps/wallet-mobile/src/legacy/config.ts]: hey @MichalSzorad please check if you can drop it.

File: apps/wallet-mobile/src/InitApp.tsx (hunk 6 above)
Comment (stackchain) [reply apps/wallet-mobile/src/InitApp.tsx]: yes, we have a specific project for dev, it's bloated but it helps sometimes.

Comment (stackchain) [reply apps/wallet-mobile/src/InitApp.tsx]: use a lower `traceSampleRate` in prod `1` can impact the other requests

**Comment by stackchain** [comment]
:arrow_up:

File: apps/wallet-mobile/src/InitApp.tsx (hunk 6 above)
Comment (michalszorad) [reply apps/wallet-mobile/src/InitApp.tsx]: @stackchain what rate would you recommend? Is 0.25 good? Can't find any recommendations on Sentry

**Comment by michalszorad** [comment]
@stackchain that's now updated

File: apps/wallet-mobile/src/InitApp.tsx (hunk 6 above)
Comment (stackchain) [reply apps/wallet-mobile/src/InitApp.tsx]: I think only 1 is problematic in prod

File: apps/wallet-mobile/src/InitApp.tsx [hunk 7]
```diff
@@ -50,3 +57,23 @@ export const initApp = async (storage: YoroiStorage) => {
...
+      tracesSampleRate: 1.0,
```
Comment (stackchain) [inline apps/wallet-mobile/src/InitApp.tsx]:
```suggestion
      tracesSampleRate: 0.25,
```

File: apps/wallet-mobile/src/legacy/config.ts (hunk 5 above)
Comment (stackchain) [inline apps/wallet-mobile/src/legacy/config.ts]: drop it, check only for the DSN.

Comment (stackchain) [reply apps/wallet-mobile/src/legacy/config.ts]: rename to SENTRY_DSN, and that is it, if it's present is enabled.

**Review by stackchain — APPROVED** [review]

**Pull request merged**